lib/internal/streams/operators.js

```diff
@@ -0,0 +1,269 @@
+'use strict';
+
+const { AbortController } = require('internal/abort_controller');
+const { isIterable } = require('internal/streams/utils');
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_ARG_VALUE,
+  }, AbortError,
+} = require('internal/errors');
+
+const {
+  NumberIsInteger,
+  Promise,
+  PromiseReject,
+  Symbol,
+} = primordials;
+
+const kEmpty = Symbol('kEmpty');
+
+async function * map(fn, options) {
+  if (typeof fn !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'fn', ['Function', 'AsyncFunction'], this);
+  }
+
+  let concurrency = 1;
+  if (options?.concurrency != null) {
+    concurrency = options.concurrency;
+  } else if (options != null && typeof options !== 'object') {
+    concurrency = options;
```

Comment:
Let's remove passing concurrency outside the object? I don't think it's particularly readable/useful

Reply:
As in, not allow `map(mapperFn, /*concurrency */ 2)`?

Reply:
Yes, allow `.map(mapper, { concurrency: 2})` which is clearer, there shouldn't be two ways to do this especially if we add another argument like `highWaterMark` which would make this sort of thing ambiguous.